### PR TITLE
add 'conflicting types for ...' matcher

### PR DESCRIPTION
**Pull request by thctamm**
Resolves issue [70](https://github.com/cs50/help50/issues/70)

### CODE REVIEW

File: helpers/clang.py [hunk 1]
```diff
@@ -144,6 +144,25 @@ def help(lines):
 
     # $ clang foo.c
     # /tmp/foo-1ce1b9.o: In function `main':
+    # foo.c:3:12: error: conflicting types for 'round'
+    # int round(int n);
+    #     ^
+    matches = re.search(r"^([^:]+):(\d+):\d+: error: conflicting types for '(.*)'", lines[0])
+    if matches:
+        response = [
+            "Looks like you're redeclaring the function {}, but with a different return type on line {} of `{}`".format(matches.group(3), matches.group(2), matches.group(1))
```
Comment (brianyu28) [inline helpers/clang.py]: Probably best to put backticks around the function name.

File: helpers/clang.py [hunk 2]
```diff
@@ -144,6 +144,25 @@ def help(lines):
...
+        ]
+        new_matches = re.search(r"^([^:]+):(\d+):\d+: note: previous declaration is here", lines[3])
```
Comment (brianyu28) [inline helpers/clang.py]: To be safe, we should check to make sure that `len(lines)` is at least 4 before trying to access `lines[3]`.

**Comment by brianyu28** [comment]
Like the other matchers, can you alphabetize this error message? Also, some of the helper sentences seem to be missing punctuation.

**Comment by thctamm** [comment]
Was already reordering it, since noticed the comment on the other PRs. You're too fast! :D

File: helpers/clang.py [hunk 3]
```diff
@@ -38,6 +38,26 @@ def help(lines):
...
+        response = [
+            "Looks like you're redeclaring the function `{}`, but with a different return type on line {} of `{}`.".format(matches.group(3), matches.group(2), matches.group(1))
+        ]
+        if len(lines) >= 4:
+            new_matches = re.search(r"^([^:]+):(\d+):\d+: note: previous declaration is here.", lines[3])
+            if new_matches:
+                if matches.group(3) == new_matches.group(3):
+                    response.append("You had already declared this function on line {}.".format(matches.group(1)))
+                else:
+                    response.append("The function `{}` is already declared in the library {}. Try renaming your function.".format(matches.group(3), new_matches.group(3).split('/')[-1])
```
Comment (brianyu28) [inline helpers/clang.py]: Did this code work when you tested it? When I just tried it, I got a syntax error— think you're missing a `)` at the end of this line.

**Comment by brianyu28** [comment]
@thctamm, could you address the merge conflicts in this PR and in your other two? Looks like some of the minor changes I made to some of the other matchers (adding/remove parentheses as needed) are conflicting with yours.

**Comment by thctamm** [comment]
Fixed the conflicts on all branches

**Comment by dmalan** [comment]
9k

**Pull request merged**